version: '3.8'

services:
  db:
    image: mysql:8.0.33
    secrets:
      - cardmaker_secret
    ports:
      - "3306:3306"
    volumes:
      - "cardmaker_db_data:/var/lib/mysql"
    networks:
      - rpiserver
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 5s
      retries: 10

  api:
    image: localhost:5000/cardmaker-api:latest
    secrets:
      - cardmaker_secret
    restart: on-failure
    volumes:
      - ./api:/app
    networks:
      - rpiserver
    healthcheck:
      test: curl --fail http://localhost:8000/tags || exit 1
      timeout: 5s
      retries: 10

  frontend:
    image: localhost:5000/cardmaker-fe:latest
    networks:
      - rpiserver
    healthcheck:
      test: curl --fail http://localhost:3000/ || exit 1
      timeout: 5s
      retries: 10

  proxy:
    image: nginx:alpine
    ports:
      - "9875:9875"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    networks:
      - rpiserver
    depends_on:
      - api
      - frontend

volumes:
  cardmaker_db_data:

networks:
  rpiserver:
    external: true

secrets:
  cardmaker_secret:
    external: true
